### PR TITLE
[QC-420] Integrate post-processing with DPL

### PR DESCRIPTION
**Pull request by knopers8**
TODO
- merge QC-419
- make "tasks", "checks", "dataSamplingPolicies" and "postprocessing" structures optional in config files
- fix the problem with DPL when two PP tasks are running and finishing one causes the other one to finish
- documentaiton

### CODE REVIEW

**Comment by knopers8** [comment]
All done, but we need https://github.com/AliceO2Group/AliceO2/pull/4506 merged first

**Comment by knopers8** [comment]
Hmm, I tihnk that double Travis appears when [WIP] is removed. The previous test stays in the list and a new one is added. I will make a fake PR to confirm.

File: Framework/src/PostProcessingDevice.cxx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+  // registering state machine callbacks
+  ctx.services().get<CallbackService>().set(CallbackService::Id::Start, [this]() { start(); });
+  ctx.services().get<CallbackService>().set(CallbackService::Id::Stop, [this]() { stop(); });
+  ctx.services().get<CallbackService>().set(CallbackService::Id::Reset, [this]() { reset(); });
+}
+
+void PostProcessingDevice::run(framework::ProcessingContext& ctx)
+{
+  // todo: enable once check runners can store the results.
```
Comment (Barthelemy) [inline Framework/src/PostProcessingDevice.cxx:55]: is this todo relevant to this class ?

**Review by Barthelemy — APPROVED** [review]

File: Framework/src/PostProcessingDevice.cxx (hunk 1 above)
Comment (knopers8) [reply Framework/src/PostProcessingDevice.cxx:55]: weird, i must have messed something up with the commits, since it is not enabled

Comment (knopers8) [reply Framework/src/PostProcessingDevice.cxx:55]: ah no, it is fine. This part is done in the next PR, at this stage we still store objects directly to the repository inside PostProcessing, because we don't send them to checkers yet. This part is correct.
